app/src/main/java/net/gsantner/opoc/frontend/GsSearchOrCustomTextDialog.java

```diff
@@ -398,31 +435,67 @@ public static void showMultiChoiceDialogWithSearchFilterUI(final Activity activi
 
+        final int padding = GsContextUtils.instance.convertDpToPx(context, 4);
         final LinearLayout titleLayout = new LinearLayout(context);
-        titleLayout.setOrientation(LinearLayout.VERTICAL);
-        titleLayout.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
+        titleLayout.setOrientation(LinearLayout.HORIZONTAL);
+        titleLayout.setPadding(4 * padding, 2 * padding, 4 * padding, padding);
+        titleLayout.setLayoutParams(new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT)
+        );
         titleLayout.setGravity(Gravity.CENTER_VERTICAL | Gravity.START);
-        titleLayout.setPadding(paddingSide, 2 * paddingBetween, paddingSide, paddingBetween);
+
+        final LinearLayout titleTextLayout = new LinearLayout(context);
+        titleTextLayout.setOrientation(LinearLayout.VERTICAL);
+        titleTextLayout.setLayoutParams(new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT, 5)
+        );
+        titleLayout.addView(titleTextLayout);
+
 
         if (dopt.titleText != 0) {
             final TextView title = new TextView(context, null, android.R.attr.windowTitleStyle);
             title.setSingleLine();
             title.setEllipsize(TextUtils.TruncateAt.END);
             title.setText(dopt.titleText);
             title.setPadding(0, 0, 0, 0);
-            titleLayout.addView(title, new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
+            titleTextLayout.addView(title, new LinearLayout.LayoutParams(
+                    LinearLayout.LayoutParams.MATCH_PARENT,
+                    LinearLayout.LayoutParams.WRAP_CONTENT));
         }
 
         if (!TextUtils.isEmpty(dopt.messageText)) {
             final TextView subTitle = new TextView(context, null, android.R.attr.textAppearanceMedium);
-            subTitle.setPadding(0, dopt.titleText == 0 ? 0 : paddingBetween, 0, 0);
+            subTitle.setPadding(0, dopt.titleText == 0 ? 0 : padding, 0, 0);
             subTitle.setText(dopt.messageText);
             subTitle.setTextIsSelectable(true);
             subTitle.setMovementMethod(LinkMovementMethod.getInstance()); // Allow links to be shown and followed
-            titleLayout.addView(subTitle, new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));
+            titleTextLayout.addView(subTitle, new LinearLayout.LayoutParams(
+                    LinearLayout.LayoutParams.MATCH_PARENT,
+                    LinearLayout.LayoutParams.WRAP_CONTENT));
+        }
+
+        if (dopt.isMultiSelectEnabled) {
+            // Using a multiple choice text view as a selectable checkbox button
```

Comment:
Adding a select all button to search or custom dialog where multi select is enabled